sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -0,0 +1,61 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SpanStatus
+import java.io.IOException
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class SentryOkHttpInterceptor(
+    private val hub: IHub = HubAdapter.getInstance()
+) : Interceptor {
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        var request = chain.request()
+
+        val url = request.url.toString()
+        val method = request.method
+
+        // read transaction from the bound scope
+        val span = hub.span?.startChild("http.client", "$method $url")
+
+        val response: Response
+
+        var code = -1
+        try {
+            span?.toSentryTrace()?.let {
+                request = request.newBuilder().addHeader(it.name, it.value).build()
+            }
+            response = chain.proceed(request)
+            code = response.code
+        } catch (e: IOException) {
+            span?.throwable = e
+            throw e
+        } finally {
+            span?.finish(SpanStatus.fromHttpStatusCode(code, SpanStatus.INTERNAL_ERROR))
+        }
+
+        val breadcrumb = Breadcrumb.http(request.url.toString(), request.method, code)
```

Comment:
wondering if it makes sense to call `Breadcrumb.http` without `code` if its -1, otherwise its adding -1 which does not make sense, or Breadcrumb.http inside of it checks if its -1, or changing it to Integer and check if its null

Reply:
👍 +1

Reply:
I changed it to always add breadcrumb even if call fails with network error.